### PR TITLE
Add a CI job to check runner det sync

### PR DESCRIPTION
**Pull request by afrittoli**
Add a new CI job that runs only when the runner determinator files are modified. The jobs checks that the runner_determinator.py script is in sync with the version embedded in _runner-determinator.yaml.

Fixes TBD

### CODE REVIEW

**Comment by afrittoli** [comment]
FYI @jeanschmidt @zxiiro @ZainRizvi @malfet

**Review by zxiiro — APPROVED** [review]

File: .github/workflows/_runner_determinator_script_sync.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: runner-determinator
```
Comment (ZainRizvi) [inline .github/workflows/_runner_determinator_script_sync.yaml:1]: nit: could you please rename the workflow file to remove the underscore prefix? The repo's convention is to use those only when a  workflow is a reusable workflow

File: .github/workflows/_runner_determinator_script_sync.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            RUNNER_DETERMINATOR_WORKFLOW_FILE=.github/workflows/_runner-determinator.yml
+            RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE=.github/scripts/runner_determinator.py
+
+            # Parse the job file, extract the script and run it, up to the final EOF,
+            # to generate the python file in the local folder
+            yq '.jobs.runner-determinator.steps[] | select(.id == "hardcode-script") | .run' \
+                "${RUNNER_DETERMINATOR_WORKFLOW_FILE}" | sed '/^EOF$/q' | bash
+
+            set +e
+            DIFF="$(diff "$(basename ${RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE})" ${RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE})"
```
Comment (ZainRizvi) [inline .github/workflows/_runner_determinator_script_sync.yaml:34]: Does this ignore that any trailing new-lines at the end of either copy of the script? (would be nice if it does!)

**Review by ZainRizvi — APPROVED** [review]
Thanks for making this change!

File: .github/workflows/_runner_determinator_script_sync.yaml (hunk 1 above)
Comment (afrittoli) [reply .github/workflows/_runner_determinator_script_sync.yaml:1]: ah, good catch, thank you

Comment (afrittoli) [reply .github/workflows/_runner_determinator_script_sync.yaml]: fixed

File: .github/workflows/_runner_determinator_script_sync.yaml (hunk 2 above)
Comment (afrittoli) [reply .github/workflows/_runner_determinator_script_sync.yaml]: `diff -B` would ignore blank line in any position within the script - @ZainRizvi would you like me to add the `-B` flag? 

Alternatively, I could parse both files with some obscure `sed` command like:

```
# Delete all trailing blank lines at end of file (only).
sed -e :a -e '/^\n*$/{$d;N;};/\n$/ba' file
```

**Comment by afrittoli** [comment]
EasyCLA check is fixed  now (not sure why it was failing).

**Review by jeanschmidt — APPROVED** [review]

**Pull request closed without merging**